src/Middleware/Diagnostics/src/StatusCodePage/StatusCodePagesExtensions.cs

```diff
@@ -210,7 +210,32 @@ public static IApplicationBuilder UseStatusCodePages(this IApplicationBuilder ap
                     context.HttpContext.Request.Path = originalPath;
                     context.HttpContext.Features.Set<IStatusCodeReExecuteFeature?>(null);
                 }
-            });
+            };
+
+            const string globalRouteBuilderKey = "__GlobalEndpointRouteBuilder";
+            // Check if UseRouting() has been called so we know if it's safe to call UseRouting()
+            // otherwise we might call UseRouting() when AddRouting() hasn't been called which would fail
+            if (app.Properties.TryGetValue("__EndpointRouteBuilder", out _) || app.Properties.TryGetValue(globalRouteBuilderKey, out _))
```

Comment:
Same idea here. Let's scope this to the WebApplicationBuiler scenario only for .NET 6:
```suggestion
            if (app.Properties.TryGetValue(globalRouteBuilderKey, out var routeBuilder))
```